presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1542,7 +1555,7 @@ else if (hasAggregates(node)) {
                     groupByExpression = groupingColumn;
                 }
 
-                Analyzer.verifyNoAggregatesOrWindowFunctions(metadata.getFunctionRegistry(), groupByExpression, "GROUP BY clause");
+                Analyzer.verifyNoAggregateWindowOrGroupingFunctions(metadata.getFunctionRegistry(), groupByExpression, "GROUP BY clause");
```

Comment:
The query error itself is a bit circular so I think it's OK. `grouping()` must reference expressions used by `GROUP BY` but `GROUP BY` references `grouping()`.